Sources/UtilityBeltNetworking/HTTPClient+Combine.swift

```diff
@@ -110,13 +110,20 @@
                                             parameters: ParameterDictionaryConvertible? = nil,
                                             headers: HTTPHeaderDictionaryConvertible? = nil,
                                             encoding: ParameterEncoding? = nil,
+                                            validators: [ResponseValidator] = [],
                                             dispatchQueue: DispatchQueue = .main,
                                             decoder: JSONDecoder = JSONDecoder()) -> AnyPublisher<T, Error> {
+            // Given that we expect to decode JSON from the response, add
+            // an additional validator that checks the appropriate mime type.
+            var validators = validators
+            validators.append(.ensureMimeType(.json))
```

Comment:
What happens if someone already added this validator, not knowing it was included by default?

I see three possible ways to avoid conflicts:

1. Make the signature `validators: [ResponseValidator] = [.ensureMimeType(.json)]`. This is a bit weird for the `Decodable` version of these requests, because we _know_ someone wouldn't want to drop them. But it would be clear they are being overridden, and it would allow someone to override in case they had another deserialization format that didn't use `JSON`. (This _is_ possible to have, such as with Firebase Firestore or GRPC, which both have `Codable` compatibility.)
2. Make the array a `Set` type. You might need to add some protocols to the `ResponseValidator` type, but should be fairly simple.
3. Check to see if the validator exists, and don't add it if so. I think to equate this you'll need to implement the same protocols as you would for #2.

Reply:
> What happens if someone already added this validator, not knowing it was included by default?

So does this matter actually?  The way the validators work, is it throws an error when the first one fails, so duplicating the `ensureMimeType` check will still ultimately fail the request as expected?

Reply:
This is also a little tricky because the `Validator` object is a struct that contains a closure.  Given closures cannot be compared, I'm not sure how we would check if a certain validation objects exist...

I'm still not too worried about this check though - the worst thing that happens is that duplicate successful validation happens on each request, which seems relatively trivial.  Is there a better way to structure validators to make them comparable?  Perhaps having a class structure instead of structs, so that we could have a subclass that can be compared (for instance a MimeTypeValidator that stores mime types and check against them?)

Reply:
@bdrelling let me know what you think we should do here when you have a sec, I think this is the only major outstanding work to be done on this PR

Reply:
We discussed all of this on a Zoom call, resolving it.